internal/provider/resource_definition_criteria_resource.go

```diff
@@ -0,0 +1,273 @@
+	var data *ResourceDefinitionCriteriaResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	httpResp, err := r.client().PostOrgsOrgIdResourcesDefsDefIdCriteriaWithResponse(ctx, r.orgId(), data.ResourceDefinitionID.ValueString(), client.PostOrgsOrgIdResourcesDefsDefIdCriteriaJSONRequestBody{
+		AppId:   optionalStringFromModel(data.AppID),
+		EnvId:   optionalStringFromModel(data.EnvID),
+		EnvType: optionalStringFromModel(data.EnvType),
+		ResId:   optionalStringFromModel(data.ResID),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(HUM_CLIENT_ERR, fmt.Sprintf("Unable to create resource definition criteria, got error: %s", err))
+		return
+	}
+
+	if httpResp.StatusCode() != 200 {
+		resp.Diagnostics.AddError(HUM_API_ERR, fmt.Sprintf("Unable to create resource definition criteria, unexpected status code: %d, body: %s", httpResp.StatusCode(), httpResp.Body))
+		return
+	}
+
+	parseResourceDefinitionCriteriaResponse(httpResp.JSON200, data)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ResourceDefinitionCriteriaResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data *ResourceDefinitionCriteriaResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	httpResp, err := r.client().GetOrgsOrgIdResourcesDefsDefIdWithResponse(ctx, r.orgId(), data.ResourceDefinitionID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(HUM_CLIENT_ERR, fmt.Sprintf("Unable to read resource definition, got error: %s", err))
+		return
+	}
+
+	if httpResp.StatusCode() == 404 {
+		resp.Diagnostics.AddWarning("Resource definition not found", fmt.Sprintf("The resource definition (%s) was deleted outside Terraform", data.ResourceDefinitionID.ValueString()))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	if httpResp.StatusCode() != 200 {
+		resp.Diagnostics.AddError(HUM_API_ERR, fmt.Sprintf("Unable to read resource definition, unexpected status code: %d, body: %s", httpResp.StatusCode(), httpResp.Body))
+		return
+	}
+
+	if httpResp.JSON200 == nil {
+		resp.Diagnostics.AddWarning("Resource definition criteria not found", fmt.Sprintf("The resource definition criteria (%s) was deleted outside Terraform", data.ID.ValueString()))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	res := *httpResp.JSON200
+
+	if res.Criteria == nil {
+		resp.Diagnostics.AddWarning("Resource definition criteria not found", fmt.Sprintf("The resource definition criteria (%s) was deleted outside Terraform", data.ID.ValueString()))
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
+	criteria := *res.Criteria
+
+	found := false
```

Comment:
this is just an observation. Looking at the API, you look into the criteria{} to identify if this is there or not.
 I guess we need an endpoint later on :eyes: all good

Reply:
Yes 👍, but that will take a bit more time.